### ISSUE TITLE
Non-conforming ID

### ISSUE THREAD

**Issue by anasyb** [balderdashy/sails-mongo#261]
Say we have a mongo db with ObjectId as a type for IDs.
Now doing this should get us an error:
```javascript
//mongo should return Error: invalid object id: **length**
User.create({id:"hellokitty"}).exec(console.log);

//mongo should return Error: invalid object id: **not hex**
User.create({id:"hellokitty00000000000000"}).exec(console.log);
```

Instead I am getting a record inserted into mongo, with `_id` initialized by mongo, and curiously *another* field `id` with NaN.

This issue could be not just particular to IDs, it *might* be also applicable to any type mismatch... not sure if it happens on attributes other than ID.

@devinivy I think regardless of the adapter, test cases should generally check to make sure that **only** the attributes that should have been created are being created. Currently to my knowledge tests are only checking if the attributes that should have been created are there and in good shape. Such tests would have caught this issue as well as issue #254 (where attributes named after associations are created in the db when they shouldn't). Just a thought.

**Comment by dogrocker** [comment]
Yes I've got the same `id = null` as you.

**Comment by akshitjain** [comment]
This issue is a major concern as when we populate the tables with the data specifying {id:xx} then it works and adapter sails-mongo understands that it corresponds to '_id' key.But when we query from the mongodb database mentioning find{id:xx} then it doesn't match it with the '_id' key.And doesn't execute the query as expected.
But when we query with '_id:xx' then it works.

**Comment by anasyb** [comment]
@akshitjain are you using mongo style hex IDs? If you're then it should work fine.... I only had issue when the ID isn't conforming to mongo style.

**Comment by akshitjain** [comment]
No i'm not using mongo style hex IDs

**Comment by sailsbot** [comment]
Thanks for posting, @anasyb.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
